### PR TITLE
Sample cleanup and upgrade

### PR DESCRIPTION
**Pull request by kalyankrishna1**
First draft with changes.Readme.md update still work in progress

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,20 +2,25 @@
...
+level: 300
+client: .NET Framework 4.5 console 
+service: .NET Framework 4.5 web api
+endpoint: AAD V1
 ---
-
 # Authenticating to Azure AD non-interactively using a username & password
+## About this sample
+### Scenario
+This sample demonstrates a .Net console application calling a web API that is secured using Azure AD. The .Net application uses the Active Directory Authentication Library (ADAL) to obtain a JWT access token through the OAuth 2.0 protocol. The access token is sent to the web API to authenticate the user. This sample shows you how to use ADAL to authenticate users via raw credentials (username and password, or Windows integrated authentication) via a text-only interface. More information is available in the ADAL.NET conceptual documentation in [Acquiring tokens with username and password](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Acquiring-tokens-with-username-and-password) and [AcquireTokenSilentAsync using Integrated authentication on Windows (Kerberos)](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/AcquireTokenSilentAsync-using-Integrated-authentication-on-Windows-(Kerberos))
```
Comment (jmprieur) [inline README.md]: @kalyankrishna1 I've added a couple of links to the conceptual documentation

File: README.md [hunk 2]
```diff
@@ -2,20 +2,25 @@
...
+### More information
 For more information about how the protocols work in this scenario and other scenarios, see [Authentication Scenarios for Azure AD](https://azure.microsoft.com/documentation/articles/active-directory-authentication-scenarios/).
 
 > Looking for previous versions of this code sample? Check out the tags on the [releases](../../releases) GitHub page.
 
 ## How To Run This Sample
 
 To run this sample you will need:
-- Visual Studio 2013
+- [Visual Studio 2017](https://aka.ms/vsdownload)
```
Comment (jmprieur) [inline README.md]: Good idea. we should do that in every sample

File: README.md [hunk 3]
```diff
@@ -37,7 +42,7 @@ There are two projects in this sample.  Each needs to be separately registered i
 3. Click on **More Services** in the left hand nav, and choose **Azure Active Directory**.
 4. Click on **App registrations** and choose **Add**.
 5. Enter a friendly name for the application, for example 'TodoListService' and select 'Web Application and/or Web API' as the Application Type. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.
-6. While still in the Azure portal, choose your application, click on **Settings** and choose **Properties**.
+6. While still in the Azure portal, choose your application (be sure that the combo-box shows **All apps**), click on **Settings** and choose **Properties**.
```
Comment (jmprieur) [inline README.md]: I've noticed that in several cases the combo-box needs to be changed to "All apps". @kalyankrishna1 did you noticed this as well?

File: README.md [hunk 4]
```diff
@@ -51,13 +56,14 @@ There are two projects in this sample.  Each needs to be separately registered i
 6. While still in the Azure portal, choose your application, click on **Settings** and choose **Properties**.
 7. Find the Application ID value and copy it to the clipboard.
 8. Configure Permissions for your application - in the Settings menu, choose the 'Required permissions' section, click on **Add**, then **Select an API**, and type 'TodoListService' in the textbox. Then, click on  **Select Permissions** and select 'Access TodoListService'.
+9. Press the **Grant Permissions** button in the bar at the top of the Required permissions blade
```
Comment (jmprieur) [inline README.md]: This did not work until I did this (see the Troubleshooting section below)

File: README.md [hunk 5]
```diff
@@ -135,10 +105,27 @@ To deploy the TodoListService to Azure Web Sites, you will create a web site, pu
...
+
+
+### Acquiring a token with Windows Integrated security
+If your PC is domain joint or AAD joint, you can also use the Windows integrated security. For this, instead of calling TextualPrompt(), you need to uncommment the line creating and instance of UserCredential without parameters: 
+See [Program.cs](https://github.com/Azure-Samples/active-directory-dotnet-native-headless/blob/update/TodoListClient/Program.cs#L159-L161)
+```C#
+UserCredential uc = new UserCredential();
+```
 
-Coming soon.
```
Comment (jmprieur) [inline README.md]: Would should not leave "Coming soon", unless we really want to do it soon. Therefore I've written a few lines on the code. @kalyankrishna1 can you please review them.

File: README.md [hunk 6]
```diff
@@ -135,10 +105,27 @@ To deploy the TodoListService to Azure Web Sites, you will create a web site, pu
...
 
-## How To Recreate This Sample
+## Troubleshooting
+- If you get the following error: ``Inner Exception : AADSTS65001: The user or administrator has not consented to use the application with ID *your app ID* named 'TodoListClient'. Send an interactive authorization request for this user and resource``, then check that you have done bullet point 9 of [Register the TodoListClient app](#Register the TodoListClient app)
 
-coming soon.
+## More information
```
Comment (jmprieur) [inline README.md]: I've added links to ADAL.NET conceptual documentation for all the features used in this sample

File: TodoListService/TodoListService.csproj [hunk 7]
```diff
@@ -44,45 +46,42 @@
...
-    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\Newtonsoft.Json.6.0.4\lib\net45\Newtonsoft.Json.dll</HintPath>
+    <Reference Include="Microsoft.Owin.Security.Jwt, Version=3.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Owin.Security.Jwt.3.0.1\lib\net45\Microsoft.Owin.Security.Jwt.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Owin.Security.OAuth, Version=3.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Owin.Security.OAuth.3.0.1\lib\net45\Microsoft.Owin.Security.OAuth.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (jmprieur) [inline TodoListService/TodoListService.csproj]: I was not sure about this one. but I checked that the sample worked. so I think we are good.

File: README.md (hunk 3 above)
Comment (kalyankrishna1) [reply README.md]: yes, I believe its something I noted in my excel sheet as well earlier

**Comment by jmprieur** [comment]
In the Readme .md
- step 9 in the creation of the service AAD app, mentions admin consent, whereas this is this user consent.

- When running the project (cloning the repo from scratch), this does not work because Visual Studio starts the url ``http://localhost:56009/`` and not ``https://localhost:44321/``. you need to change the ``IISUrl`` in ``TodoListService.csproj`` to  ``<IISUrl>https://localhost:44321/</IISUrl>``
- when that's done, this still does not work as the client receives a 401. Looking at the debugger, it seems that ``ConfigureAuth(IAppBuilder app)`` is never called (which would explain)

Minor remark. it seems jquery's version in this new code is lower than the one in the previous code. Not a problem, though.

**Comment by jmprieur** [comment]
LGTM, and the sample now works

**Pull request merged**